### PR TITLE
fix(rust): Fix overflow in substring

### PR DESCRIPTION
**Pull request by rluvaton-flarion**
The error happen when running fuzzy testing
```
 ./target/release/flarion-cli run-fuzz-test --verbose --spark-master "local" --seed 29884 --number-of-files 10 --number-of-rows 200 --number-of-columns 100 --number-of-queries 1000
```

### CODE REVIEW

File: crates/polars-core/src/chunked_array/ops/flarion_funcs.rs [hunk 1]
```diff
@@ -43,7 +43,8 @@ pub fn flarion_substring(s: &str, from: i32, len: i32) -> String {
         let mut found_start = false;
 
         let distance_to_advance = from.unsigned_abs() as usize;
-        let end_idx = std::cmp::max(0, distance_to_advance as i32 - len) as usize; // We know len is positive
+
+        let end_idx = (distance_to_advance as i32).wrapping_sub(len) as usize; // We know len is positive
```
Comment (shanyfl) [inline crates/polars-core/src/chunked_array/ops/flarion_funcs.rs]: Wrapping sub guarantee proper indexing?

Comment (rluvaton-flarion) [reply crates/polars-core/src/chunked_array/ops/flarion_funcs.rs]: wrapping sub is the same as before but without panic

Comment (rluvaton-flarion) [reply crates/polars-core/src/chunked_array/ops/flarion_funcs.rs]: i was wrong, it's not working

Comment (rluvaton-flarion) [reply crates/polars-core/src/chunked_array/ops/flarion_funcs.rs]: fixed

**Comment by EmilyFlarionIO** [comment]
Did this pass fuzz/integration?

**Review by shanyfl — APPROVED** [review]

**Pull request closed without merging**